cmd/backup.go

```diff
@@ -0,0 +1,140 @@
+		availableStorer := []string{}
+		for _, s := range datastoreRepo.GetAll() {
+			availableStorer = append(availableStorer, s.Name())
+		}
+		var storerName string
+		if err := survey.AskOne(&survey.Select{
+			Message: "Select supported datastores?",
+			Options: availableStorer,
+		}, &storerName); err != nil {
+			return err
+		}
+
+		var qs = []*survey.Question{
+			{
+				Name: "name",
+				Prompt: &survey.Input{
+					Message: "What is the resource name?",
+					Help:    "Input urn of the resource",
+				},
+				Validate: survey.ComposeValidators(validateNoSlash, survey.MinLength(3),
+					survey.MaxLength(1024)),
+			},
+			{
+				Name: "description",
+				Prompt: &survey.Input{
+					Message: "Why is this backup needed?",
+					Help:    "Describe intention to help identify the backup",
+				},
+				Validate: survey.ComposeValidators(survey.MinLength(3)),
+			},
+			{
+				Name: "backupDownstream",
+				Prompt: &survey.Confirm{
+					Message: "Backup downstream?",
+					Help:    "Select yes to also backup the downstream resources",
+				},
+			},
+		}
+		inputs := map[string]interface{}{}
+		if err := survey.Ask(qs, &inputs); err != nil {
+			return err
+		}
+		resourceName := inputs["name"].(string)
+		description := inputs["description"].(string)
+		backupDownstream := inputs["backupDownstream"].(bool)
+		backupRequest := &pb.BackupDryRunRequest{
+			ProjectName:      project,
+			Namespace:        namespace,
+			ResourceName:     resourceName,
+			DatastoreName:    storerName,
+			Description:      description,
+			IgnoreDownstream: !backupDownstream,
+		}
+
+		return runBackupDryRunRequest(l, conf, backupRequest)
+	}
+	return backupCmd
+}
+
+func runBackupDryRunRequest(l log.Logger, conf config.Provider, backupRequest *pb.BackupDryRunRequest) (err error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+
+	var conn *grpc.ClientConn
+	if conn, err = createConnection(dialTimeoutCtx, conf.GetHost()); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Info("can't reach optimus service")
+		}
+		return err
+	}
+	defer conn.Close()
+
+	requestTimeoutCtx, requestCancel := context.WithTimeout(context.Background(), replayTimeout)
+	defer requestCancel()
+
+	l.Info("please wait...")
+	runtime := pb.NewRuntimeServiceClient(conn)
+
+	backupDryRunResponse, err := runtime.BackupDryRun(requestTimeoutCtx, backupRequest)
+	if err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			l.Info("backup dry run took too long, timing out")
+		}
+		return errors.Wrapf(err, "request failed to backup job %s", backupRequest.ResourceName)
+	}
+
+	printBackupDryRunResponse(l, backupRequest, backupDryRunResponse)
+	return nil
+}
+
+func printBackupDryRunResponse(l log.Logger, backupRequest *pb.BackupDryRunRequest, backupResponse *pb.BackupDryRunResponse) {
+	l.Info("Backup list for %s\n\n", coloredNotice(backupRequest.ResourceName))
```

Comment:
logger doesn't support taking %s you might need to use `fmt.Sprintf` here and we remove one `\n` I think.

Reply:
have updated this. i also updated the message, you might want to recheck it.